HW1

VitalyKapustin-git opened the pull request:
- added controlles
- addes models
- passed postman tests

VitalyKapustin-git committed:
- done

ddddemo1984 commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,25 @@
+@Data
+@Builder
+public class Film {
+    int id;
```
Нужно везде добавить модификаторы доступа

ddddemo1984 commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,25 @@
+import java.time.LocalDate;
+
+@Data
+@Builder
```
Не нашёл где используется builder

VitalyKapustin-git committed:
feat:
- added tests for controllers via MockMvc;

ddddemo1984 commented on src/test/java/ru/yandex/practicum/filmorate/controllers/FilmControllerTest.java:
```diff
@@ -0,0 +1,112 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
```
Весь неиспользуемый импорт нужно удалить

ddddemo1984 commented on src/main/java/ru/yandex/practicum/filmorate/controllers/FilmController.java:
```diff
@@ -0,0 +1,73 @@
+                        film.getId() < 1
+        ) {
+            log.warn("Произошла ошибка при обновлении фильма: {}", film);
+            throw new ValidationException("Произошла ошибка при обновлении фильма");
```
Такой текст ошибки не информативен, клиент твоего реста не поймёт чего от него хотят. Сделай, пожалуйста, отдельное сообщение об ошибке для каждой проверки. Тогда в тесте сможешь проверить, что например на старый фильм у тебя контроллер ругается именно на это.
Чтобы не дублировать код, при добавлении и обновлении, схожие проверки можешь реализовать в общем методе validate

VitalyKapustin-git committed:
feat:
- added personal tests for each bordered situation;
- added tests for bordered situations;

ddddemo1984 commented on src/test/java/ru/yandex/practicum/filmorate/controllers/FilmControllerTest.java:
```diff
@@ -0,0 +1,145 @@
+                "......................................................................" +
+                "......................................................................" +
+                "......................................................................");
+        assertThrows(ValidationException.class, () -> filmController.filmBasicValidation(film1));
```
Теперь ты сделал для каждого случая свой текст ошибки, но не проверяешь его в тесте. Я имел в виду что-то вроде: 
Exception exToTest = assertThrows(ValidationException.class, () -> new FilmController().filmBasicValidation(film1));
        assertEquals("Произошла ошибка при обновлении фильма (описанее более 200 символов)", exToTest.getMessage());

Поправь, пожалуйста для всех тестов, чтобы убедиться что падает именно на том, что ожидаешь

VitalyKapustin-git committed:
refactor:
- changed tests from assertTrows to assertEquals to check exception message
